[windows] remove pyc files during rollback

derekwbrown opened the pull request:
### What does this PR do?

If the install fails, some PYC files were being left around (Those generated by the
agent before it failed).  Remove them during rollback so that we end up with a clean
system.

Re-enables kitchen tests verifying.

derekwbrown committed:
Fix windows build.  Change to allow specifying 2, 3 or both included
in build had a couple of problems.

in datadog-a7, windows command had faulty --install-option flag
Also, modified build to conditionally include the right two/three DLL
components in MSI (build fails otherwise)

derekwbrown committed:
Manually clean up PYC files on rollback, so that we don't leave anything
around on failure.

Re-enable kitchen test that verifies this behavior

olivielpeau commented on omnibus/resources/agent/msi/cal/delfiles.cpp:
```diff
@@ -0,0 +1,70 @@
+    if (wcscmp(str, L".") && wcscmp(str, L"..")) return FALSE;
+    return TRUE;
+}
+BOOL DeleteDirectory(const TCHAR* sPath) {
```
also, does the name of this function actually reflect what the function does? (If I understand correctly it deletes `.pyc` files and their parent directories if they're empty, so name like `DeletePycFiles` would be more descriptive IMO)

olivielpeau reviewed (commented):
Left a few inline comments. Nice to see that we can re-enable the test again! 👍 

I believe the merge conflicts come from the shared commit with https://github.com/DataDog/datadog-agent/pull/3549, I'll let you update this PR accordingly :)

derekwbrown committed:
Fix recursive delete function
Fix path to search
Add new parameters to localbuild/parameters.wxi (local build impact
only)

derekwbrown committed:
Fix kitchen templates to use older chef when doing local/hyper-v runs.
No impact on CI; but newer chef seems to break the hyper-v run.

Change install to not supply log parameter (done automatically now)
revert failure check to ensure that the whole DataDog directory is gone

derekwbrown committed:
Merge remote-tracking branch 'origin/master' into db/fix_pyc_installfail

derekwbrown committed:
REmove outdated comment

olivielpeau commented on omnibus/resources/agent/msi/localbuild/parameters.wxi:
```diff
@@ -8,4 +8,6 @@
   <?define EtcFiles="C:/omnibus-ruby/src\etc\datadog-agent" ?>
   <?define ResourcesDir="c:\dev\go\src\github.com\datadog\datadog-agent\omnibus\resources\agent\msi"?>
+  <?define IncludePython2= "true" ?>
+  <?define IncludePython3= "false" ?>
```
why are these 2 lines included in the PR?

derekwbrown replied:
For anyone who wants to be able to do a test MSI without going through a whole `omnibus` build.  The whole `localbuild` directory exists to be able to do an MSI  test->change->build cycle without waiting for an omnibus build.

olivielpeau reviewed (dismissed):
Only reviewed the logic of the deletion at a high level, overall it looks good to me, if you were able to confirm it works well in practice let's merge.

Left a comment on a couple of lines (for which I don't understand why they're in the diff), let me know

truthbk commented on omnibus/resources/agent/msi/cal/CustomAction.cpp:
```diff
@@ -368,6 +368,7 @@ extern "C" UINT __stdcall DoUninstall(MSIHANDLE hInstall) {
     return WcaFinalize(er);
 }
 
+BOOL DeleteFilesInDirectory(const wchar_t* dirname, const wchar_t* ext);
```
Nit: since `DeleteFilesInDirectory` is defined in a different `cpp`, from a project standpoint I think it might make more sense to have this in one of the header files (or a new one), and include it, as opposed to a forward declaration here.

derekwbrown replied:
it's not a nit... it was lazy.  Fixed.

truthbk reviewed (commented):
Looks good to me, just a nit with a forward declaration.

derekwbrown committed:
Fix forward declaration

truthbk reviewed (approved):
Thanks for the fix! :)
